mesonbuild/dependencies.py

```diff
@@ -721,8 +724,16 @@ def get_link_args(self):
             args.append('-L' + os.path.join(self.boost_root, 'lib'))
         for module in self.requested_modules:
             module = BoostDependency.name2lib.get(module, module)
-            if module in self.lib_modules or module in self.lib_modules_mt:
-                linkcmd = '-lboost_' + module
+            libname = 'boost_' + module
+            # The compiler's library detector is the most reliable so use that first.
+            default_detect = self.cpp_compiler.find_library(libname, self.environment, [])
+            if default_detect is not None:
+                emon_args = self.cpp_compiler.find_library('boost_test_exec_monitor')
```

Comment:
This will add it unconditionally and repeatedly for each module; is this what we want?